crates/xilem_html/src/view.rs

```diff
@@ -0,0 +1,219 @@
+// Copyright 2023 the Druid Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Integration with xilem_core. This instantiates the View and related
+//! traits for DOM node generation.
+
+use std::{any::Any, borrow::Cow, ops::Deref};
+
+use xilem_core::{Id, MessageResult};
+
+use crate::{context::Cx, ChangeFlags};
+
+mod sealed {
+    pub trait Sealed {}
+}
+
+// A possible refinement of xilem_core is to allow a single concrete type
+// for a view element, rather than an associated type with a bound.
+/// This trait is implemented for types that implement `AsRef<web_sys::Node>`.
+/// It is an implementation detail.
+pub trait DomNode: sealed::Sealed {
+    fn into_pod(self) -> Pod;
+    fn as_node_ref(&self) -> &web_sys::Node;
+}
+
+impl<N: AsRef<web_sys::Node> + 'static> sealed::Sealed for N {}
+impl<N: AsRef<web_sys::Node> + 'static> DomNode for N {
+    fn into_pod(self) -> Pod {
+        Pod(Box::new(self))
+    }
+
+    fn as_node_ref(&self) -> &web_sys::Node {
+        self.as_ref()
+    }
+}
+
+/// This trait is implemented for types that implement `AsRef<web_sys::Element>`.
+/// It is an implementation detail.
+pub trait DomElement: DomNode {
+    fn as_element_ref(&self) -> &web_sys::Element;
+}
+
+impl<N: DomNode + AsRef<web_sys::Element>> DomElement for N {
+    fn as_element_ref(&self) -> &web_sys::Element {
+        self.as_ref()
+    }
+}
+
+/// A trait for types that can be type-erased and impl `AsRef<Node>`. It is an
+/// implementation detail.
+pub trait AnyNode: sealed::Sealed {
+    fn as_any_mut(&mut self) -> &mut dyn Any;
+
+    fn as_node_ref(&self) -> &web_sys::Node;
+}
+
+impl<N: AsRef<web_sys::Node> + Any> AnyNode for N {
+    fn as_any_mut(&mut self) -> &mut dyn Any {
+        self
+    }
+
+    fn as_node_ref(&self) -> &web_sys::Node {
+        self.as_ref()
+    }
+}
+
+impl sealed::Sealed for Box<dyn AnyNode> {}
+impl DomNode for Box<dyn AnyNode> {
+    fn into_pod(self) -> Pod {
+        Pod(self)
+    }
+
+    fn as_node_ref(&self) -> &web_sys::Node {
+        self.deref().as_node_ref()
+    }
+}
+
+/// A container that holds a DOM element.
+///
+/// This implementation may be overkill (it's possibly enough that everything is
+/// just a `web_sys::Element`), but does allow element types that contain other
+/// data, if needed.
+pub struct Pod(pub Box<dyn AnyNode>);
+
+impl Pod {
+    fn new(node: impl DomNode) -> Self {
+        node.into_pod()
+    }
+
+    fn downcast_mut<'a, T: 'static>(&'a mut self) -> Option<&'a mut T> {
```

Comment:
lifetime `'a` can be omitted (I think this is somewhere on main as well though)

Reply:
Good find. I did a `rg "'a"` and this is the only one I found.